src/core/src/core_logic/PatchInstaller.py

```diff
@@ -177,14 +178,48 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         self.last_still_needed_packages = all_packages
         self.last_still_needed_package_versions = all_package_versions
 
+        stopwatch_for_batch_install_process = Stopwatch(self.env_layer, self.telemetry_writer, self.composite_logger)
+        stopwatch_for_batch_install_process.start()
+
+        installed_update_count, patch_installation_successful, maintenance_window_batch_cutoff_reached = self.install_packages_in_batches(all_packages, packages, package_versions, maintenance_window, package_manager)
+
+        stopwatch_for_batch_install_process.stop()
+
+        # Check for packages available and install them in batches. Some packages may not be installed due to:
+        # (a) Not enough time remaining in maintenance window
+        # (b) Failure during package installation
+        # (c) New packages added during the brief time batch installation was in progress
+        packages, package_versions = self.get_remaining_packages_to_install(package_manager)
+
+        batch_processing_perf_log  = {Constants.PerfLogTrackerParams.TASK: "Installation of packages in batches", "Installed packages count in batch processing": str(installed_update_count), "number of remaining packages to install": str(len(packages)), 
+                                      Constants.PerfLogTrackerParams.PATCH_OPERATION_SUCCESSFUL: str(patch_installation_successful), "Is installation in batches stopped due to not enough remaining time to install in batches": str(maintenance_window_batch_cutoff_reached)}
+
+        stopwatch_for_batch_install_process.write_telemetry_for_stopwatch(str(batch_processing_perf_log))
+
+        if len(packages) == 0:
+            installed_update_count = self.log_metrics_and_perform_final_reconciliation(package_manager, maintenance_window, patch_installation_successful, False, installed_update_count)
+            return installed_update_count, patch_installation_successful, False
```

Comment:
return maintenance_window_exceeded instead of return False as it would be easy to read code.
The value of maintenance_window_exceeded  is always False at this point.

Reply:
Done